baybe/utils/basic.py

```diff
@@ -121,31 +123,54 @@ def to_tuple(x: Sequence) -> tuple:
     return tuple(x)
 
 
-def filter_attributes(
+def match_attributes(
     object: Any,
     callable_: Callable,
-    ignore: Collection[str] = ("self", "kwargs", "args"),
-) -> dict[str, Any]:
+    /,
+    strict: bool = True,
+) -> tuple[dict[str, Any], dict[str, Any]]:
     """Find the attributes of an object that match with a given callable signature.
 
     Parameters appearing in the callable signature that have no match with the given
     object attributes are ignored.
 
     Args:
-        object: The object whose attributes are to be returned.
+        object: The object whose attributes are to be matched.
         callable_: The callable against whose signature the attributes are to be
             matched.
-        ignore: A collection of parameter names to be ignored in the signature.
+        strict: If ``True``, an error is raised when the object has attributes that
+            are not found in the callable signature. If ``False``, the these
+            attributes are returned separately.
+
+    Raises:
+        ValueError: If applied to a non-attrs object.
+        UnmatchedAttributeError: In strict mode, if not all attributes can be matched.
 
     Returns:
-        A dictionary mapping the matched attribute names to their values.
+        * A dictionary mapping the matched attribute names to their values.
```

Comment:
Is this rendered properly in the documentation`?

Reply:
yes, displayed as a bullet list